packages/react-badge/Spec.md

```diff
@@ -133,14 +133,56 @@ A Presence Badge represents someone's availbility or status
 
 A Counter Badge is a visual indicator for numeric values such as tallies and scores.
 
-#### API
+```typescript
+/**
+ * {@docCategory CounterBadge}
+ */
+export type CounterBadgeColors = 'accent' | 'warning' | 'important' | 'severe' | 'informative';
+
+/**
+ * {@docCategory CounterBadge}
+ */
+export interface CounterBadgeProps extends Omit<BadgeProps, 'appearance' | 'shape'> {
+  /**
+   * A Badge can be circular or rounded
+   * @defaultvalue circular
+   */
+  shape?: Extract<BadgeShape, 'rounded' | 'circular'>;
```

Comment:
I believe the inline values will be useful for `Badge` I'm not entirely sure if we consider that to be the 'superset'

I believe Extract's utility here is keeping the different options consistent. so my vision is smth like this:

```typescript
shape?: Extract<BadgeProps['shape'], 'rounded' | 'circular'>
```

WDYT ?

Reply:
I am actually fine with any of the solutions. So let's just decide which one.

Reply:
Honestly any of these works and you're free to pick whichever makes the most sense to you @assuncaocharles.

That said, after thinking about it more I personally like a third option. This is a nontrivial change from what you have, so I understand if you'd prefer to leave things as-is... but here's the details in case you'd like to go this route:

Don't have any of the badges inherit from each other. Instead, move the shared functionality/styling that's common to all badges into hooks that can be used by all of the badges. Each badge would use the shared hooks, and then add anything unique to that badge. This improves separation of concerns, and doesn't introduce weirdness where one badge inherits from another but omits some inherited functionality.

* `useBaseBadgeState` would handle all of the state management for the `icon` slot, and defaults for common props (size, etc). (This could be templated on the `TBadgeState` type, so it could accept default props and return the correct state type).
* `useBaseBadgeStyles` would define styles for the `icon` slot, plus sizing, rounded corners, etc.
* `renderBaseBadge` would render the slots. Since this is identical for all badges, it could be re-exported by each badge as `renderBadge`/`renderCounterBadge`/`renderPresenceBadge`. (Re-exporting allows for flexibility in the future to rewrite those functions if they need to diverge).
* Then each badge would define its own props and hooks that internally call the base hooks, then add their own functionality.
  * `Badge` would add the extra colors and appearances that are only used by it.
  * `CounterBadge` would add the count prop, and the status colors unique to `CounterBadge`, etc.
  * `PresenceBadge` would add icons and styling for the presence statuses, etc.

Reply:
For now I will update with Ling strategy:

```
/**
   * A Badge can be circular or rounded
   * @defaultvalue circular
   */
  shape?: Extract<BadgeProps['shape'], 'rounded' | 'circular'>;

  /**
   * A Badge can be filled, ghost
   * @defaultvalue filled
   */
  appearance?: Extract<BadgeProps['appearance'], 'filled' | 'ghost'>;
```